### PR TITLE
FIX: handle zero division in heatmap when all SHAP values are zero

### LINKED ISSUE #4448: BUG: heatmap plot renders incorrectly when all SHAP values are zero
### Issue Description

In `shap/plots/_heatmap.py`, two division operations fail silently when all SHAP values are zero.

First occurrence (line 158):
```
fx = values.T.sum(0)
ax.plot(
    -fx / np.abs(fx).max() - 1.5,  # np.abs(fx).max() = 0 when all SHAP values are zero
    color="#000000",
    linewidth=1,
)
```

Second occurrence (line 166):
```
(feature_values / np.abs(feature_values).max()) * values.shape[0] / 20
# np.abs(feature_values).max() = 0 for the same reason
```
When all SHAP values are zero, both denominators evaluate to 0, producing:
`RuntimeWarning: invalid value encountered in divide`

This causes two visual artifacts:

- The `f(x)` line at the top of the heatmap is blank instead of rendering as a flat line.

- The bar plot on the right spine is blank instead of rendering zero-width bars.

Image

### Minimal Reproducible Example

```python
import numpy as np
import shap
from shap import Explanation

exp = Explanation(
    values=np.zeros((20, 5)),
    base_values=np.zeros(20),
    data=np.zeros((20, 5)),
    feature_names=[f"feature_{i}" for i in range(5)],
)
shap.plots.heatmap(exp, show=False)
```

### Traceback

```shell
/usr/local/lib/python3.12/dist-packages/shap/plots/_heatmap.py:158: RuntimeWarning: invalid value encountered in divide
  -fx / np.abs(fx).max() - 1.5,
/usr/local/lib/python3.12/dist-packages/shap/plots/_heatmap.py:166: RuntimeWarning: invalid value encountered in divide
  (feature_values / np.abs(feature_values).max()) * values.shape[0] / 20,
```

### Expected Behavior

The plot should render cleanly with no warnings. Specifically:

- The `f(x)` line should render as a flat horizontal line (since all SHAP values 
  are zero, every instance has identical model output, and a flat line is the correct 
  representation)
- The bar plot on the right spine should render zero-width bars
- No `RuntimeWarning` should be raised

**Expected fix:**

For the `f(x)` line:
```python
fx = values.T.sum(0)
fx_max = np.abs(fx).max()
if fx_max > 0:
    fx_normalized = fx / fx_max
else:
    fx_normalized = fx  # already zeros, flat line, no division needed
ax.plot(-fx_normalized - 1.5, color="#000000", linewidth=1)
```

For the bar plot:
```python
fv_max = np.abs(feature_values).max()
bar_widths = (
    (feature_values / fv_max) * values.shape[0] / 20
    if fv_max > 0
    else np.zeros_like(feature_values)
)
```

This fix only affects the degenerate zero case. Normal non-zero inputs 
are completely unaffected.

### Bug report checklist

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [x] I'd be interested in making a PR to fix this bug

### Installed Versions

`0.51.0`

AI Usage: The zero-division possibility was found by me. Validation for that issue was done by Claude. I have understood the fixes.

### PR DESCRIPTION
**Pull request by BaibhavKundu2005**
## What does this PR do?
Fixes two division by zero operations in `shap/plots/_heatmap.py` that 
occur when all SHAP values are zero.

## What problem does it solve?
Closes #4448

When all SHAP values are zero, both `np.abs(fx).max()` and 
`np.abs(feature_values).max()` evaluate to `0`, causing:
- `RuntimeWarning: invalid value encountered in divide` (twice)
- Blank `f(x)` line at the top of the heatmap
- Blank bar plot on the right spine

## Fix
Guard both divisions with a zero-check. When the max is zero, skip 
normalization since the values are already zero and no scaling is needed.

For the `f(x)` line:
- If `fx_max > 0`: normalize as before
- If `fx_max == 0`: use zeros directly, renders as a flat line

For the bar plot:
- If `fv_max > 0`: normalize as before  
- If `fv_max == 0`: use zero-width bars

This fix only affects the degenerate zero case. All non-zero inputs 
are completely unaffected.

## Checklist
- [x] I have read CONTRIBUTING.md
- [x] Branch is up to date with master
- [x] Pre-commit hooks pass
- [x] Unit test added: `test_heatmap_all_zero_shap_values`

### CODE REVIEW

**Comment by BaibhavKundu2005** [comment]
@CloseChoice Hi, just a gentle ping whenever you have a moment. I have opened a PR #4449 to address the zero-division error and incorrect rendering of plots when all SHAP values are zero. All CI checks have passed. I'm an ESoC 2026 Batch 2 applicant contributing to SHAP as part of my evaluation - happy to make any changes if needed. Thanks!

File: tests/plots/test_heatmap.py [hunk 1]
```diff
@@ -25,3 +25,25 @@ def test_heatmap_feature_order(explainer):
     )
     plt.tight_layout()
     return fig
+
+
+def test_heatmap_all_zero_shap_values():
```
Comment (CloseChoice) [inline tests/plots/test_heatmap.py:30]: thanks for the test

**Review by CloseChoice — APPROVED** [review]
This looks good to me. This is a rare edge case but a valid one. Pls ping on green pipelines.

EDIT: forgot to mention that I checked that the test fails on master.

**Comment by BaibhavKundu2005** [comment]
@CloseChoice Thank you for the review and the approval! Just pinging as requested. All CI pipelines are green and the checks have successfully passed. Let me know if you need anything else from my end before merging.

**Comment by BaibhavKundu2005** [comment]
@CloseChoice Just a gentle ping. The CI checks have passed. Let me know if anything else needs to be done before merging. Thanks!

**Comment by CloseChoice** [comment]
@BaibhavKundu2005 thanks for the PR
